File: src/find_the_parity_outlier/js/findOutlier.js
function findOutlier(integers){
    let evenArr = integers.filter(f => f % 2 === 0);
    let result;
    evenArr.length === 1 ? result = evenArr : result = integers.filter(f => f % 2 !== 0);
    return result[0];
}

module.exports = findOutlier;

File: src/kyu_6/find_the_parity_outlier/java/FindOutlier.java
package kyu_6.find_the_parity_outlier.java;

import java.util.Arrays;

public class FindOutlier {
    static int find(int[] integers) {
        int evenCounter = 0;
        int oddCounter = 0;
        int result = 0;
        for (int i : integers) {
            int check = i % 2 == 0 ? evenCounter++ : oddCounter++;
        }
        if (evenCounter < oddCounter) {
            for (int i : integers) {
                if (i % 2 == 0) {
                    result = i;
                }
            }
        } else {
            for (int i : integers) {
                if (i % 2 != 0) {
                    result = i;
                }
            }
        }
        return result;
    }
}
